core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -751,6 +751,16 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .setScope(Scope.SERVER)
           .build();
+  public static final PropertyKey UNDERFS_OBJECT_STORE_SKIP_PARENT_DIRECTORY_CREATION =
+      new Builder(Name.UNDERFS_OBJECT_STORE_SKIP_PARENT_DIRECTORY_CREATION)
+          .setDefaultValue(true)
+          .setDescription("Do not create parent directory for new files. Object stores generally "
+              + "uses prefix which is not required for creating new files. Skipping parent "
+              + "directory is recommended for better performance. Set this to false if the "
+              + "object store requires prefix creation for new files.")
```

Comment:
Does any object store require prefix creation as you know?

Reply:
I am not aware of any. The S3 APIs only uses prefix and there is no need to create a prefix before creating an object under the prefix.

Reply:
we can do more testing to make sure that